pkg/jx/cmd/get_vault_config.go

```diff
@@ -0,0 +1,81 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/jenkins-x/jx/pkg/jx/cmd/templates"
+	"github.com/jenkins-x/jx/pkg/jx/cmd/vault"
+	"github.com/spf13/cobra"
+	"gopkg.in/AlecAivazis/survey.v1/terminal"
+	"io"
+	"runtime"
+)
+
+type GetVaultConfigOptions struct {
+	GetOptions
+
+	Namespace string
+}
+
+var (
+	getVaultConfigLong = templates.LongDesc(`
+		Echoes the configuration required for connecting to a vault using the official vault CLI client	
+	`)
+
+	getVaultConfigExample = templates.Examples(`
+		# Gets vault config
+		jx get vault-config
+	`)
+)
+
+// NewCmdGetVaultConfig creates a new command for 'jx get secrets'
+func NewCmdGetVaultConfig(f Factory, in terminal.FileReader, out terminal.FileWriter, errOut io.Writer) *cobra.Command {
+	options := &GetVaultConfigOptions{
+		GetOptions: GetOptions{
+			CommonOptions: CommonOptions{
+				Factory: f,
+				In:      in,
+				Out:     out,
+				Err:     errOut,
+			},
+		},
+	}
+
+	cmd := &cobra.Command{
+		Use:     "vault-config",
+		Short:   "Gets the configuration for using the official vault CLI client",
+		Long:    getVaultConfigLong,
+		Example: getVaultConfigExample,
+		Run: func(c *cobra.Command, args []string) {
+			options.Cmd = c
+			options.Args = args
+			err := options.Run()
+			CheckErr(err)
+		},
+	}
+
+	options.addGetFlags(cmd)
+
+	cmd.Flags().StringVarP(&options.Namespace, "namespace", "n", "", "Namespace from where to get the vault config")
+	cmd.Flags().StringVarP(&options.Namespace, "name", "m", "", "Name of the vault to get the config for")
+	return cmd
+}
+
+// Run implements the command
+func (o *GetVaultConfigOptions) Run() error {
+	clientFactory := vault.VaultClientFactory{
+		Options: o,
+	}
+	client, err := clientFactory.NewVaultClient(o.Namespace)
+	if err != nil {
+		return err
+	}
+
+	// Echo the client config out to the command line to be piped into bash
+	if runtime.GOOS == "windows" {
```

Comment:
I would use `os.Setenv` function instead of handling specific per OS.

Reply:
That's not quite the same. `os.Setenv` will set the env parameter for the current (`jx`) process, which will get discarded when the process terminates. This needs to be able to set the user's environment for use with the official `vault` CLI. Unfortunately (given environment variables get dicarded when the current running process exits) the only way to do this is to export a string that the user can pipe into the current terminal. 
This is the way that AWS tools do `docker` config (and the easiest way to do it) - see https://docs.aws.amazon.com/cli/latest/reference/ecr/get-login.html